frontend/src/scenes/plugins/edit/PluginDrawer.tsx

```diff
@@ -158,9 +177,16 @@ export function PluginDrawer(): JSX.Element {
                                     ) : null}
                                     {fieldConfig.type ? (
                                         <Form.Item
-                                            label={fieldConfig.name || fieldConfig.key}
+                                            label={
+                                                fieldConfig.type !== 'choice' || isValidChoiceConfig(fieldConfig)
+                                                    ? fieldConfig.name || fieldConfig.key
+                                                    : ''
+                                            }
                                             extra={
-                                                fieldConfig.hint ? (
+                                                fieldConfig.hint &&
+                                                (fieldConfig.type === 'choice'
+                                                    ? isValidChoiceConfig(fieldConfig)
```

Comment:
To clean this up, one approach is to make a helper:

```
const isValidField = fieldConfig => fieldConfig.type !== 'choice' || isValidChoiceConfig(fieldConfig)
```

or something like that and use that before the `{fieldConfig.type ? ( <Form.Item ...` code to either show an error or the form item.